tools/fix_auth/models.rb

```diff
@@ -10,6 +10,12 @@ class FixAuthentication < ActiveRecord::Base
     self.inheritance_column = :_type_disabled
   end
 
+  class FixConfigurationScript < ActiveRecord::Base
+    include FixAuth::AuthModel
+    self.table_name = "configuration_scripts"
+    self.password_columns = %w[credentials]
+  end
```

Comment:
Lets leverage AuthConfigModel instead

```suggestion
  class FixConfigurationScript < ActiveRecord::Base
    include FixAuth::AuthConfigModel
    self.table_name = "configuration_scripts"
    self.password_columns = %w[credentials]

    # no particular fields for passwords, instead encrypt everything
    self.password_fields = []
    # blank prefix matches all fields
    self.password_prefix = ""
  end
```

(fun fact: `fix_auth` is flexible and allows columns to be declared that are removed from the database. This is necessary since this has been used across manageiq versions and columns come and go. Using `serialize` in the models caused issues when columns are dropped, so we manually serialize in `recrypt`.)

For this new code to support `jsonb`, we need to optionally serialize the column:


```
# ConfigurationAuthModel

      # if old_value is a hash, then we assume serialization is already handled
      # else it is probably a string and we want to yaml serialize/deserialize
      def recrypt(old_value, options = {})
        hash = old_value.kind_of?(Hash) ? old_value : YAML.load(old_value)

        Vmdb::SettingsWalker.walk(hash) do |key, value, _path, owning|
          owning[key] = super(value, options) if password_field?(key) && value.present?
        end

        symbol_keys ? hash.deep_symbolize_keys! : hash.deep_stringify_keys!
        old_value.kind_of?(Hash) ? hash : hash.to_yaml
     end
```

https://github.com/agrare/manageiq/compare/fix_auth_configuration_script_credentials...kbrock:manageiq:fix_auth_configuration_script_credentials